cmd/ecloud/ecloud_vpnendpoint_test.go

```diff
@@ -0,0 +1,430 @@
+package ecloud
+
+import (
+	"errors"
+	"testing"
+
+	gomock "github.com/golang/mock/gomock"
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/assert"
+	"github.com/ukfast/cli/internal/pkg/clierrors"
+	"github.com/ukfast/cli/test/mocks"
+	"github.com/ukfast/cli/test/test_output"
+	"github.com/ukfast/sdk-go/pkg/service/ecloud"
+)
+
+func Test_ecloudVPNEndpointList(t *testing.T) {
+	t.Run("DefaultRetrieve", func(t *testing.T) {
+		mockCtrl := gomock.NewController(t)
+		defer mockCtrl.Finish()
+
+		endpoint := mocks.NewMockECloudService(mockCtrl)
+
+		endpoint.EXPECT().GetVPNEndpoints(gomock.Any()).Return([]ecloud.VPNEndpoint{}, nil).Times(1)
+
+		ecloudVPNEndpointList(endpoint, &cobra.Command{}, []string{})
+	})
+
+	t.Run("MalformedFlag_ReturnsError", func(t *testing.T) {
+		mockCtrl := gomock.NewController(t)
+		defer mockCtrl.Finish()
+
+		endpoint := mocks.NewMockECloudService(mockCtrl)
+		cmd := &cobra.Command{}
+		cmd.Flags().StringArray("filter", []string{"invalidfilter"}, "")
+
+		err := ecloudVPNEndpointList(endpoint, cmd, []string{})
+
+		assert.IsType(t, &clierrors.ErrInvalidFlagValue{}, err)
+	})
+
+	t.Run("GetVPNEndpointsError_ReturnsError", func(t *testing.T) {
+
+		mockCtrl := gomock.NewController(t)
+		defer mockCtrl.Finish()
+
+		endpoint := mocks.NewMockECloudService(mockCtrl)
+
+		endpoint.EXPECT().GetVPNEndpoints(gomock.Any()).Return([]ecloud.VPNEndpoint{}, errors.New("test error")).Times(1)
+
+		err := ecloudVPNEndpointList(endpoint, &cobra.Command{}, []string{})
+
+		assert.Equal(t, "Error retrieving VPN endpoints: test error", err.Error())
+	})
+}
+
+func Test_ecloudVPNEndpointShowCmd_Args(t *testing.T) {
+	t.Run("ValidArgs_NoError", func(t *testing.T) {
+		err := ecloudVPNEndpointShowCmd(nil).Args(nil, []string{"vpns-abcdef12"})
+
+		assert.Nil(t, err)
+	})
+
+	t.Run("InvalidArgs_Error", func(t *testing.T) {
+		err := ecloudVPNEndpointShowCmd(nil).Args(nil, []string{})
+
+		assert.NotNil(t, err)
+		assert.Equal(t, "Missing VPN endpoint", err.Error())
+	})
+}
+
+func Test_ecloudVPNEndpointShow(t *testing.T) {
+	t.Run("SinglePolicy", func(t *testing.T) {
+		mockCtrl := gomock.NewController(t)
+		defer mockCtrl.Finish()
+
+		endpoint := mocks.NewMockECloudService(mockCtrl)
+
+		endpoint.EXPECT().GetVPNEndpoint("vpns-abcdef12").Return(ecloud.VPNEndpoint{}, nil).Times(1)
```

Comment:
tests should use `vpne-` for vpn endpoints

Reply:
Good spot, sorted :)